torch/cuda/graphs.py

```diff
@@ -232,8 +240,18 @@ def __init__(
 
     def __enter__(self) -> None:
         # Free as much memory as we can for the graph
-        torch.cuda.synchronize()
-        gc.collect()
+
+        # Originally we unconditionally garbage collected here. On one hand
```

Comment:
I'd rather have an explicit argument controlling gc, rather than rely on unpredictable non-deterministic once-every-10s collection. Also, it's better  torch.cuda.synchronize unconditionally

Reply:
@ngimel I updated to add a config to control the behavior instead of a clock.